### DISCUSSION TITLE
LaTeX as a single source

### DISCUSSION THREAD

**Discussion by uihsnv** [singlesourcepub/community#17] [Ideas]
Since LaTeX was created specifically for the purpose of typesetting documents, does it not make sense to it as the primary "single source" instead of Markdown or HTML?

**Comment by makomi** [comment]
That is an interesting question :)

AFAIK there are two types of document description languages for two different use cases:
- The one type allows you to exactly describe how the final result shall look like, e.g. LaTeX.
- The other type only lets you declare the type of content (heading, emphasis, link) but does not allow you to control how it will be rendered on the user's monitor, e.g. Markdown, HTML.

I don't have any experience with converting Markdown to nice looking PDFs but @tarleb should know the limitations. I specifically wonder whether you can reach journal quality typesetting levels with Markdown to PDF conversion. IIRC it has been done, but I don't remember the limitations and how difficult it is.

**Reply by uihsnv** [reply]
Well, that isn't entirely true: one could use the stylistic prescriptions that LaTeX allows you to make, more as suggestions. That is up-to the engine that renders the source into the final document, just like an HTML/CSS rendering engine can make its own decisions.

All I'm saying, is that as far as a single "source" document is concerned, we already have LaTeX that allows us to make all the specifications that we need.

It is much much more detailed than Markdown, and definitely more user-friendly than HTML. I feel like, amongst that options that we currently have, it strikes the best balance and is optimal for the task.

**Reply by makomi** [reply]
Yeah, maybe it would work to use LaTeX annotations and ignore the explicit ones for document types that do not require them. However, I don't know how many people feel comfortable using LaTeX instead Markdown. If that is a concern, it would be nice to have the option to write documents in either Markdown or LaTeX. Then everybody can choose depending on whether their primary publication will be a journal or a website.

**Reply by tarleb** [reply]
> I specifically wonder whether you can reach journal quality typesetting levels with Markdown to PDF conversion.

We've done that here: http://kommunikation-gesellschaft.de/

The input was Markdown which was converted to both HTML and PDF (via LaTeX). We had to "cheat" in some cases, as Markdown tables were too simple for complex tables; we resolved it by embedding raw LaTeX and HTML tables in the input.

**Comment by tarleb** [comment]
It all has pros and cons, and the following is just my limited experience.

IMHO, LaTeX is still unmatched when it comes to typesetting and flexibility. It remains a safe choice when producing print publications, or even online-only PDFs. But the power and flexibility of LaTeX is also its biggest downside. The more advanced the input, the less straight-forward it becomes to convert the document into other formats like HTML. E.g., TeX makes it easy to combine two glyphs by using negative spacing, but that's not easy to translate into other, less graphically powerful formats. And with LaTeX's focus on graphical representations, authors are often tempted to focus on that, and not the semantic meaning.

This could all be solved by defining a subset of LaTeX which can safely be used in a single-source publishing environment. The focus would be on semantics, not typesetting details. Being limited to that would take away a lot of what makes LaTeX great, but it would indeed allow it to be used for SSP.

One other thing that's currently a blocker is the incomplete support for accessible PDFs. HTML and EPUB are, more or less, accessible by default, while tagging is still not readily available for PDFs produced through LaTeX.

**Comment by jcolomb** [comment]
I see 2 problems with latex:
- you always end up tweaking the latex commands to get the output you want, this takes forever…
- it is very difficult to get a team to work on a latex document, even using overleaf (my boss just sent me an annotated docx file he obtained from the latex doc…)

on a wider scale, latex primary outputs are pdf, while ssp will need some xml output to be compatible with the current ecosystem.

one can also see the problem of using latex by browsing some latex templates on overleaf: each journal gets different ways to implement authors and affiliations, in most cases on ly to look like the final pdf, but without any of the metadata needed (meaning the info are requested again in forms for real submissions)

**Reply by uihsnv** [reply]
None of those are issues with latex as a 'source'. It is still the best 'semantic' prescription.
